### PR TITLE
[CI] fix virtualenv version to deflake linux://python/ray/tests:test_runtime_env_complicated

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

Looks the virtualenv has been upgraded between the success and failed test.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rickyyx — APPROVED** [review]
Oh wow , nice. Is there a bad version from virtualenv?

**Review by rkooo567 — APPROVED** [review]
This means we won’t work with that version of virtualenv? Maybe we should create another issue to fix that?

File: python/requirements.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ pyyaml
 aiosignal
 frozenlist
 requests
-virtualenv>=20.0.24
+virtualenv>=20.0.24, != 20.21.1
```
Comment (jjyao) [inline python/requirements.txt:21]: Is it a virtualenv bug? Should we do `<20.21.1`?

Comment (jjyao) [reply python/requirements.txt:21]: The reasoning is that if it's not their issue then the next virtualenv will also break things.

Comment (scv119) [reply python/requirements.txt:21]: i've no idea. let's first see if this deflakes it.

**Review by jjyao — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Pull request merged**